PyRDF/Proxy.py

```diff
@@ -2,6 +2,9 @@
 from .backend import Executor
 from .CallableGenerator import CallableGenerator
 
+Backend_env = "local"
```

Comment:
I believe these two variables should not be defined in `Proxy`. `Proxy` just imports them from the entity that manages them. That should be some module called `Backend.py`, which actually could be your current `Executor.py`.
Summary: I would rename `Executor.py` to `Backend.py` and define these two variables there, and import them from `Proxy.py`.

Reply:
Just to be fully precise, `backend_env` and `backend_conf` would be class data members of `Backend`.

Reply:
@etejedor , it'd work only if I declare those variables as global variables inside `Backend.py`. Class data members will get reset to the default values when you invoke it from a different module. I'd initially kept `Backend_env` and  `Backend_conf` as class data members of Proxy class. If I change the value of these members inside `___init.py__` , it wouldn't work as I'm importing Proxy class separately everywhere else. So, are global variables fine inside `Backend.py` ?

Reply:
So what about having this:
- A superclass `Backend`, implemented in `Backend.py`, which has a constructor that receives a `config` object and assigns it to an instance data member called `config`. It also has a (virtual) method `execute` with this signature:
```
def execute(callable):
  raise NotImplementedError()
```
- Subclasses (e.g. `Local`, implemented in `Local.py`) which inherit from `Backend` and override the `execute` method with the right behaviour in each case.
- `Proxy` defines a class data member called `backend`, initially `None`
- `PyRDF/__init__.py`, in `use`, constructs a backend object of the right type (e.g. `Local`) with the `config` received from the user (or the default one) and assigns that backend object to `Proxy.backend`.

Reply:
@etejedor , sounds like a good solution to me. One question, where would the `execute()` function of `Local` and `Dist` be called. Inside `execute()` of parent class ?

Reply:
No, `execute` would be called from `Proxy` on a `backend` object that has been previously injected in `Proxy` (read above). That `backend` object can be e.g. a `Local`, so the `Local` implementation of `execute` will be invoked. Note that the `execute` of the parent class is simulating a virtual method: it raises a exception when you try to call it.

Reply:
@etejedor , works :+1: